modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java

```diff
@@ -0,0 +1,1144 @@
+import com.intuit.wasabi.tests.model.factory.EventFactory;
+import com.intuit.wasabi.tests.model.factory.ExperimentFactory;
+import com.intuit.wasabi.tests.model.factory.UserFactory;
+import com.jayway.restassured.response.Response;
+
+import static com.intuit.wasabi.tests.library.util.Constants.*;
+
+/**
+ * RollUp integration tests
+ */
+public class RollUpIntegrationTest extends TestBase {
+
+	private static final String FROM_TIME = "fromTime";
+	private static final String QBO = "qbo";
+	protected static final String RED = "red";
+	protected static final String BLUE = "blue";
+
+	protected static final String PROD = "PROD";
+	protected static final String QA = "QA";
+
+	private String yesterday;
+	private String today;
+	private String tomorrow;
+
+	private Experiment experiment;
+	private List<Bucket> buckets = new ArrayList<>();
+	private String[] labels = { BLUE, RED };
+	private double[] allocations = { .50, .50, };
+	private boolean[] control = { false, true };
+	private User userBill = UserFactory.createUser("Bill");
+	private User userJane = UserFactory.createUser("Jane");
+	private User userTom = UserFactory.createUser("Tom");
+
+	private User[] users = { userBill, userJane, userTom };
+
+	private String actionImpression = "IMPRESSION";
+	private String actionClick = "click";
+	private String actionLoveIt = "love it";
+	private SimpleDateFormat dateFormat;
+	private String tomorrowPlus3;
+
+	private Connection connection;
+
+	/**
+	 * Initializes a default experiment.
+	 */
+	public RollUpIntegrationTest() throws Exception {
+
+		setResponseLogLengthLimit(1000);
+
+		dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+		Calendar cal = Calendar.getInstance();
+		cal.add(Calendar.DATE, -1);
+		yesterday = dateFormat.format(cal.getTime());
+		yesterday += "T00:00:00+0000";
+
+		cal = Calendar.getInstance();
+		cal.add(Calendar.DATE, 0);
+		today = dateFormat.format(cal.getTime());
+		today += "T00:00:00+0000";
+
+		cal = Calendar.getInstance();
+		cal.add(Calendar.DATE, 1);
+		tomorrow = dateFormat.format(cal.getTime());
+		tomorrow += "T00:00:00+0000";
+
+		cal = Calendar.getInstance();
+		cal.add(Calendar.DATE, 4);
+		tomorrowPlus3 = dateFormat.format(cal.getTime());
+		tomorrowPlus3 += "T00:00:00+0000";
+
+		experiment = ExperimentFactory.createExperiment();
+		experiment.startTime = yesterday;
+		experiment.endTime = tomorrowPlus3;
+		experiment.samplingPercent = 1.0;
+		experiment.label = "experiment";
+		experiment.applicationName = QBO + UUID.randomUUID();
+
+		DefaultNameExclusionStrategy experimentComparisonStrategy = new DefaultNameExclusionStrategy(
+				"creationTime", "modificationTime", "ruleJson");
+		experiment.setSerializationStrategy(experimentComparisonStrategy);
+
+	}
+
+	@BeforeClass
+	public void prepareDBConnection() throws Exception {
+		Class.forName("com.mysql.jdbc.Driver").newInstance();
+		connection = DriverManager.getConnection(
+				appProperties.getProperty("database.url"), "readwrite",
```

Comment:
Readwrite is also retrievable from the properties in this PR (see properties file below).